### PR TITLE
fix(core): fix #4180 by adding validity check for minWidth and maxWidth

### LINKED ISSUE #4180: Columns not displaying properly
Hi,

when I am setting minWidth property for the columns they are not displayed properly.Below is the plunker to replicate this issue : 

http://plnkr.co/edit/OraYwr4YyWsDNcFOqRZ7?p=preview

Please look into this

Thanks,
Navneet

### PR DESCRIPTION
**Pull request by merlinchen**
- String and number are both allowed.

- Valid String value will parse into number properly.

- Invalid String value will throw an error.

### CODE REVIEW

File: src/js/core/factories/GridColumn.js [hunk 1]
```diff
@@ -464,8 +464,22 @@ angular.module('ui.grid')
...
-    self.minWidth = !colDef.minWidth ? 30 : colDef.minWidth;
-    self.maxWidth = !colDef.maxWidth ? 9000 : colDef.maxWidth;
+    ['minWidth', 'maxWidth'].forEach(function (name) {
+      var minOrMaxWidth = colDef[name];
+      var parseErrorMsg = "Cannot parse column " + name + " '" + minOrMaxWidth + "' for column named '" + colDef.name + "'";
+
+      if (!angular.isString(minOrMaxWidth) && !angular.isNumber(minOrMaxWidth)) {
+        self[name] = ((name === 'minWidth') ? 30 : 9000);
+      } else if (angular.isString(minOrMaxWidth)) {
+        if (minOrMaxWidth.match(/^(\d+)$/)) {
```
Comment (JLLeitschuh) [inline src/js/core/factories/GridColumn.js]: Can you add a comment describing this regex?

File: src/js/core/factories/GridColumn.js [hunk 2]
```diff
@@ -464,8 +464,22 @@ angular.module('ui.grid')
...
+          self[name] = parseInt(minOrMaxWidth.match(/^(\d+)$/)[1], 10);
```
Comment (JLLeitschuh) [inline src/js/core/factories/GridColumn.js]: Can you add a comment describing this regex?

File: src/js/core/factories/GridColumn.js [hunk 3]
```diff
@@ -464,8 +464,22 @@ angular.module('ui.grid')
       }
     }
 
-    self.minWidth = !colDef.minWidth ? 30 : colDef.minWidth;
-    self.maxWidth = !colDef.maxWidth ? 9000 : colDef.maxWidth;
+    ['minWidth', 'maxWidth'].forEach(function (name) {
```
Comment (JLLeitschuh) [inline src/js/core/factories/GridColumn.js]: Should we also check the `width` variable too?

File: src/js/core/factories/GridColumn.js [hunk 4]
```diff
@@ -464,8 +464,22 @@ angular.module('ui.grid')
...
+      var minOrMaxWidth = colDef[name];
+      var parseErrorMsg = "Cannot parse column " + name + " '" + minOrMaxWidth + "' for column named '" + colDef.name + "'";
+
+      if (!angular.isString(minOrMaxWidth) && !angular.isNumber(minOrMaxWidth)) {
+        self[name] = ((name === 'minWidth') ? 30 : 9000);
```
Comment (JLLeitschuh) [inline src/js/core/factories/GridColumn.js]: Can you add a comment line describing this logic?
For example:
Sets default minwidth and maxwidth values

File: src/js/core/factories/GridColumn.js (hunk 2 above)
Comment (merlinchen) [inline src/js/core/factories/GridColumn.js]: The regex is a copy paste from the `width` variable checking logic.

File: src/js/core/factories/GridColumn.js (hunk 3 above)
Comment (merlinchen) [inline src/js/core/factories/GridColumn.js]: The `width` variable is already checked. 0_0
[https://github.com/merlinchen/ui-grid/blob/col-min-and-max-width-validity-check/src/js/core/factories/GridColumn.js#L435](https://github.com/merlinchen/ui-grid/blob/col-min-and-max-width-validity-check/src/js/core/factories/GridColumn.js#L435)

File: src/js/core/factories/GridColumn.js (hunk 4 above)
Comment (merlinchen) [inline src/js/core/factories/GridColumn.js]: Sure.

**Comment by JLLeitschuh** [comment]
This looks good. Squash it into one commit and I'd have no problem merging it in.

**Comment by merlinchen** [comment]
Squashed.

**Pull request merged**